platforms/chibios/drivers/wear_leveling/wear_leveling_efl.c

```diff
@@ -66,10 +69,14 @@ bool backing_store_init(void) {
 
 #else // defined(WEAR_LEVELING_EFL_FIRST_SECTOR)
 
-    // Work out how many sectors we want to use, working backwards from the end of the flash
+        // Work out how many sectors we want to use, working backwards from the end of the flash
+#    if defined(WEAR_LEVELING_PROTECT_LAST_SECTORS)
```

Comment:
this should ideally always be 
```c
#define WEAR_LEVELING_PROTECT_LAST_SECTORS 1
```

however, I left it configurable. Going to need some feedback ( it will also need to go in docs, dd config etc)

Reply:
Why would this be `1`? There's no need to always protect the last sector?

Reply:
You might want it for other MCUs but it should not be the default.

Reply:
I should had phrased it better. What I meant here was that from my POV, if there's a need to protect something it should only be the last sector ( 1 sector). So, while the flag adds flexibility on *how many sectors to guard at the end of flash*, it might be unnecessary. Not *defaulting to guarding all MCUs*.

So, the question is if we need that configuration flag or is it better to be hardcoded to `1` without it.

Reply:
So couldn't you just add this to the top of the file, and then not need conditional compilation below?

```c
#ifndef WEAR_LEVELING_PROTECTED_SECTOR_COUNT
#define WEAR_LEVELING_PROTECTED_SECTOR_COUNT 0
#endif // WEAR_LEVELING_PROTECTED_SECTOR_COUNT
```

...and:

```c
flash_sector_t last_sector = desc->sectors_count - WEAR_LEVELING_PROTECTED_SECTOR_COUNT;
```

Reply:
meaning, configuration logic being

`#define WEAR_LEVELING_PROTECT_LAST_SECTOR`
vs
`#define WEAR_LEVELING_PROTECT_LAST_SECTORS num_sectors`